### DISCUSSION TITLE
Abstract file system?

### DISCUSSION THREAD

**Discussion by pshinners-blz** [OpenAssetIO/OpenAssetIO#46] [General]
I see similar projects like USD AR2 and Houdini's File helpers making a focused point of abstracting away file systems. I haven't seen similar features in OAIO yet. Did I miss this when reviewing the code? This feels like a crucial requirement for a modern asset api library.

**Comment by foundrytom** [comment]
100% on this.

We saw file abstraction as a separate topic though. OAIO is, in essence, about turning a URI into an URL. You can then independently map the resolved URL's protocol to a way get the data - exactly in the way that the Houdini File System API does. We've been talking to them about the possibilities of making that more generally available. We didn't want to re-invent any wheels here. Felt like it was a better separation of concerns if OAIO was a layer over the top, rather than conflating the two topics. Then you could also have the benefit of file abstraction without being forced to use OAIO.

Does the Houdini File System feel like a good solution for this topic?
